test/install_test.go

```diff
@@ -299,48 +305,88 @@ func TestDashboard(t *testing.T) {
 }
 
 func TestInject(t *testing.T) {
-	var out string
-	var err error
-
-	prefixedNs := TestHelper.GetTestNamespace("smoke-test")
-
-	out, err = testutil.ReadFile("testdata/smoke_test.yaml")
+	resources, err := testutil.ReadFile("testdata/smoke_test.yaml")
 	if err != nil {
 		t.Fatalf("failed to read smoke test file: %s", err)
 	}
-	err = TestHelper.CreateNamespaceIfNotExists(prefixedNs, map[string]string{
-		k8s.ProxyInjectAnnotation: k8s.ProxyInjectEnabled,
-	})
-	if err != nil {
-		t.Fatalf("failed to create %s namespace: %s", prefixedNs, err)
-	}
 
-	out, err = TestHelper.KubectlApply(out, prefixedNs)
-	if err != nil {
-		t.Fatalf("kubectl apply command failed\n%s", out)
-	}
+	testCases := []struct {
+		ns          string
+		autoinject  bool
+		annotations map[string]string
+	}{
+		{
+			ns:         nsSmokeAuto,
+			autoinject: true,
+			annotations: map[string]string{
+				k8s.ProxyInjectAnnotation: k8s.ProxyInjectEnabled,
+			},
+		},
+		{
+			ns:         nsSmokeManual,
+			autoinject: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("%s", tc.ns), func(t *testing.T) {
+			var out string
+
+			prefixedNs := TestHelper.GetTestNamespace(tc.ns)
+
+			err := TestHelper.CreateNamespaceIfNotExists(prefixedNs, tc.annotations)
+			if err != nil {
+				t.Fatalf("failed to create %s namespace: %s", prefixedNs, err)
+			}
 
-	for _, deploy := range []string{"smoke-test-terminus", "smoke-test-gateway"} {
-		err = TestHelper.CheckPods(prefixedNs, deploy, 1)
-		if err != nil {
-			t.Fatalf("Unexpected error: %v", err)
-		}
-	}
+			out, err = TestHelper.KubectlApply(resources, prefixedNs)
+			if err != nil {
+				t.Fatalf("kubectl apply command failed\n%s", out)
+			}
 
-	url, err := TestHelper.URLFor(prefixedNs, "smoke-test-gateway", 8080)
-	if err != nil {
-		t.Fatalf("Failed to get URL: %s", err)
-	}
+			if !tc.autoinject {
+				cmd := []string{"inject", "--manual", "testdata/smoke_test.yaml"}
 
-	output, err := TestHelper.HTTPGetURL(url)
-	if err != nil {
-		t.Fatalf("Unexpected error: %v %s", err, output)
-	}
+				var injectReport string
+				out, injectReport, err = TestHelper.LinkerdRun(cmd...)
+				if err != nil {
+					t.Fatalf("linkerd inject command failed: %s\n%s", err, out)
+				}
+
+				err = TestHelper.ValidateOutput(injectReport, "inject.report.golden")
+				if err != nil {
+					t.Fatalf("Received unexpected output\n%s", err.Error())
+				}
+
+				out, err = TestHelper.KubectlApply(out, prefixedNs)
```

Comment:
this deploys a second time for the test case, consider modifying the code above so we only call `KubectlApply` once per test?